src/main/cpp/src/decimal_utils.cu

```diff
@@ -520,6 +519,109 @@ inline __device__ int precision10(chunked256 value) {
     return -1;
 }
 
+__device__ chunked256 set_scale_and_round(chunked256 data, int old_scale, int new_scale) {
+  if (old_scale != new_scale) {
+    if (new_scale < old_scale) {
+      int const raise = old_scale - new_scale;
+      int const multiplier = pow_ten(raise).as_128_bits();
```

Comment:
I just realized in the general case that raise and/or drop could be much larger than `pow_ten` supports. This should be a very extreme corner case that would require negative scale (positive scale in CUDF) decimal values which spark has off by default. In those cases I think it would happen if pone side is so much smaller than the other side that adding or subtracting them would be a NOOP, so we might want to special case it inside of the DecimalAdd or DecimalSubtract operator.  We probably want to check for this case before ever calling these kernels and have it either do the right thing, or throw an error on the CPU before we ever reach the GPU.

Reply:
Sounds to me, that this special case will only affect add/subtract. 
Will division/multiplication be also affected? 

Regardless of above, can/should we try to fix this here or create a follow-on?

Reply:
This should be handled on the plugin side?

Reply:
That is fine, but I would feel better if we documented it in the javadocs for the methods and ideally had some checks on the CPU before calling this, so we don't hit the assert.

Reply:
I don't think I understand your original comment. The max scale supported by Spark is 38, and the min is -38. Considering that the `min_scale = -38` and the raise or drop will be 76. Which case will make the raise or drop be more than 76 for `pow_ten` to hit the assert?

Reply:
I stand corrected. Thanks @revans2 
`Seq("1E100", "5E100").toDF.withColumn("dec", col("value").cast(DecimalType(38, -100))).show()`